sunbeam-python/sunbeam/provider/local/steps.py

```diff
@@ -255,3 +254,286 @@ def _update_microcluster_status(self, status: dict, microcluster_status: dict):
             node_status["clusterd-status"] = microcluster_status.get(node_name, {}).get(
                 "status"
             )
+
+
+def sriov_questions():
+    return {
+        "configure_sriov": sunbeam.core.questions.ConfirmQuestion(
+            "Configure SR-IOV?",
+            default_value=False,
+            description=(
+                "This allows specifying a list of SR-IOV devices that "
+                "will be exposed to Openstack instances."
+            ),
+        ),
+    }
+
+
+class LocalConfigSRIOVStep(BaseStep):
+    """Prompt user for SR-IOV configuration."""
+
+    def __init__(
+        self,
+        client: Client,
+        node_name: str,
+        jhelper: JujuHelper,
+        model: str,
+        manifest: Manifest | None = None,
+        accept_defaults: bool = False,
+        show_initial_prompt: bool = True,
+    ):
+        super().__init__("SR-IOV Settings", "Configure SR-IOV")
+        self.client = client
+        self.node_name = node_name
+        self.jhelper = jhelper
+        self.model = model
+        self.manifest = manifest
+        self.accept_defaults = accept_defaults
+        self.variables: dict = {}
+        # Avoid the "Configure SR-IOV?" question if the user
+        # specifically asked for this.
+        self.show_initial_prompt = show_initial_prompt
+
+    def prompt(
+        self,
+        console: Console | None = None,
+        show_hint: bool = False,
+    ) -> None:
+        """Determines if the step can take input from the user.
+
+        Prompts are used by Steps to gather the necessary input prior to
+        running the step. Steps should not expect that the prompt will be
+        available and should provide a reasonable default where possible.
+        """
+        if not console:
+            LOG.info("No console available, skipping prompt.")
+            return
+
+        self.variables = sunbeam.core.questions.load_answers(
+            self.client, PCI_CONFIG_SECTION
+        )
+
+        pci_whitelist: list[dict] = []
+        excluded_devices: dict[str, list] = {}
+
+        if self.manifest:
+            if (
+                self.manifest.core.config.pci
+                and self.manifest.core.config.pci.device_specs
+            ):
+                pci_whitelist = self.manifest.core.config.pci.device_specs
+                LOG.debug("PCI whitelist from manifest: %s", pci_whitelist)
+            if (
+                self.manifest.core.config.pci
+                and self.manifest.core.config.pci.excluded_devices
+            ):
+                excluded_devices = self.manifest.core.config.pci.excluded_devices
```

Comment:
nitpick: this is a mouthful, could maybe reduce to:

```suggestion
            pci_config = self.manifest.core.config.pci
            if (
                pci_config
                and pci_config.device_specs
            ):
                pci_whitelist = pci_config.device_specs
                LOG.debug("PCI whitelist from manifest: %s", pci_whitelist)
            if (
                pci_config
                and pci_config.excluded_devices
            ):
                excluded_devices = pci_config.excluded_devices
```

Reply:
Makes sense, thanks.